lib/spack/spack/environment/environment.py

```diff
@@ -2634,6 +2634,36 @@ def _ensure_env_dir():
 
     shutil.copy(envfile, target_manifest)
 
+    # Copy relative path includes that live inside the environment dir
+    try:
+        manifest = EnvironmentManifestFile(environment_dir)
+    except Exception:
+        # error handling for bad manifests is handled on other code paths
+        return
+
+    includes = manifest[TOP_LEVEL_KEY].get("include", [])
+    for include in includes:
+        if os.path.isabs(include):
+            continue
+
+        abspath = pathlib.Path(os.path.normpath(environment_dir / include))
+        common_path = pathlib.Path(os.path.commonpath([environment_dir, abspath]))
+        if common_path != environment_dir:
+            # Warn that we are not copying relative path
+            msg = "Spack will not copy relative include path from outside environment"
+            msg += f" directory: {include}"
+            tty.warn(msg)
```

Comment:
Warning seems too string here. If you are copying from outside the environment, it's likely you meant to, and it's an assumption of your env.  I would make this either debug or just a message.